internal/util/unregisterer.go

```diff
@@ -18,6 +18,22 @@ func WrapWithUnregisterer(reg prometheus.Registerer) *Unregisterer {
 	}
 }
 
+func isUncheckedCollector(c prometheus.Collector) bool {
+	descChan := make(chan *prometheus.Desc, 10)
+
+	go func() {
```

Comment:
Starting a goroutine here seems a bit dodgy. Can you achieve the same with select and default in a loop?

Reply:
I cannot think of an alternative way :/ The Prometheus Go client uses the same kind of code with a buffered channel of size 10 and a goroutine. You never know how many descriptions there will be, and `c.Describe(descChan)` won't return until it sent all those descriptions down a channel.

Reply:
Yep, it's in the comment above the function, above line 37 in this file.